features/sdk-node-properties.js

```diff
@@ -0,0 +1,247 @@
+const escape = require('escape-html');
+
+const isPropertyKey = (key) => key.startsWith('.');
+const propertyKeyName = (key) => key.substring(1);
+
+const specificationPointRegExp = /^[A-Z]{1,3}[1-9]\d*([a-z]((([1-9]\d*)?[a-z])?([1-9]\d*)?)?)?$/;
+
+class SpecificationPoint {
+  constructor(value) {
+    if (!specificationPointRegExp.test(value)) {
+      throw new Error(`Value '${value}' is not formatted like a specification point.`);
+    }
+    this.value = value;
+  }
+
+  toString() {
+    return this.value;
+  }
+
+  toHtmlLink() {
+    const url = `https://docs.ably.com/client-lib-development-guide/features/#${this.value}`;
+    return `<a href="${url}" target="_blank" rel="noopener"><code>${escape(this.value)}</code></a>`;
+  }
+}
+
+const smithyShapeIdentifiers = createMapOfSmithyShapeIdentifiers();
+
+/**
+ * Creates a map of Smithy shape identifiers.
+ *
+ * @returns {Map} Smithy shape identifiers and the regular expressions to be used to qualify
+ * that strings match them, where entry keys are Smithy shape identifiers as a `string` and the
+ * values are `RegExp` instances.
+ * @see https://awslabs.github.io/smithy/1.0/spec/core/model.html#shape-id
+ */
+function createMapOfSmithyShapeIdentifiers() {
+  const identifierStart = '(_*[A-Za-z])';
+  const identifierChars = '[A-Za-z\\d_]';
+  const identifier = `(${identifierStart}${identifierChars}*)`;
+  const namespace = `(${identifier}+(\\.${identifier})*)`;
+  return new Map([
+    ['shape_id_member', new RegExp(`^\\$${identifier}$`)],
+    ['absolute_root_shape_id', new RegExp(`^${namespace}#${identifier}$`)],
+
+    // Our 'shape_id' is intentionally less inclusive than that defined in
+    // Smithy's version 1.0 Shape ID ABNF. We have ignored the intermediate
+    // 'root_shape_id', ignoring the fact that it can be either an
+    // 'absolute_root_shape_id' or an 'identifier', insisting that we expect
+    // to see a namespace. We can always consider opening this up in future
+    // if we find that some SDK manifests need to have identifiers at root
+    // which don't have a namespace component.
+    ['shape_id', new RegExp(`^${namespace}#${identifier}\\$${identifier}$`)],
+  ]);
+}
+
+/**
+ * Look up and return the Smithy identifier matching the form of the given value.
+ *
+ * @param {string} value The value to be inspected and matched to a shape identifier.
+ * @returns {string} The shape idenfier that matches the value.
+ * @throws When a shape identifier could not be matched to the value.
+ */
+function matchSmithyShapeIdentifier(value) {
```

Comment:
Can never return `false`. It returns `true` or throws.

Reply:
Oops, good spot. It works, but as a side effect - `found` should not be a boolean. Once my nominated review window has closed for this PR then I shall fix this. Please leave this conversation until I have done that, thanks.

Proposed fix:

- rename `found` to `foundIdentifier`
- initialise `foundIdentifier` with `undefined` (not `false`)

Reply:
Could be written as this (unless you are intentionally looking for the last available match) 

```js
for(let [regExp, shapeIdentifier] of smithyShapeIdentifiers.entries()){
  if(value.match(regExp)){
    return shapeIdentifier;
  }
}
throw new Error(`...`);
```

Reply:
Resolving this conversation on @tomkirbygreen's behalf as the code being commented on as been removed from scope of this PR.